src/main/ruby/core/file.rb

```diff
@@ -1126,8 +1126,10 @@ def self.utime(atime, mtime, *paths)
     end
     paths.each do |path|
       path = Rubinius::Type.coerce_to_path(path)
-      n = POSIX.truffleposix_utimes(path, atime.to_i, atime.usec,
-                                          mtime.to_i, mtime.usec)
+      atime_usec = atime.kind_of?(Time) ? atime.usec : 0;
+      mtime_usec = mtime.kind_of?(Time) ? mtime.usec : 0;
+      n = POSIX.truffleposix_utimes(path, atime.to_i, atime_usec,
+                                          mtime.to_i, mtime_usec)
```

Comment:
Should the conversion be done earlier, for example with `Rubinius::Type.coerce_to`?
What does MRI do?

Reply:
Hmm. MRI looks at the `atime` and `mtime` arguments and converts them to a `Time` by effectively using `Time.at` if they aren't already a `Time`. I'll change the code to do that once before looping over the paths. We're still missing some cases that MRI will accept, but that's more a hole in our implementation of `Time.at` and would probably fall over on real file operations anyway.

Reply:
> We're still missing some cases that MRI will accept

Which cases specifically?